homeassistant/components/mqtt/light/schema_json.py

```diff
@@ -205,7 +305,14 @@ def _setup_from_config(self, config: ConfigType) -> None:
         self._attr_supported_features |= (
             config[CONF_EFFECT] and LightEntityFeature.EFFECT
         )
-        if not self._config[CONF_COLOR_MODE]:
+        if supported_color_modes := self._config.get(CONF_SUPPORTED_COLOR_MODES, []):
```

Comment:
Why do we default to an empty list here? I think it should just be:
```suggestion
        if supported_color_modes := self._config.get(CONF_SUPPORTED_COLOR_MODES):
```